modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/bundle/BatchEngineBundleTracker.java

```diff
@@ -158,55 +139,19 @@ public Bundle addingBundle(Bundle bundle, BundleEvent bundleEvent) {
 				}
 			}
 
-			_batchEngineUnitProcessor.processBatchEngineUnits(
-				singleCompanyBatchEngineUnits);
+			boolean alreadyProcessed = _isAlreadyProcessed(bundle);
+
+			if (!alreadyProcessed) {
```

Comment:
Question, what if the `singleCompanyBatchEngineUnits` is empty? Do we process it?

Reply:
Something like:

```
boolean alreadyProcessed = _isAlreadyProcessed(bundle);

if (!singleCompanyBatchEngineUnits.isEmpty() && !alreadyProcessed) {
	_batchEngineUnitProcessor.processBatchEngineUnits(
		singleCompanyBatchEngineUnits);
}

if (!multiCompanyBatchEngineUnits.isEmpty()) {
	_multiCompanyBatchEngineUnitProcessor.registerBatchEngineUnits(
		bundle, multiCompanyBatchEngineUnits, !alreadyProcessed);
}
```